sda-download/api/sda/sda.go

```diff
@@ -29,6 +33,43 @@ func sanitizeString(str string) string {
 	return pattern.ReplaceAllString(str, "[identifier]: $1")
 }
 
+func reencryptHeader(oldHeader []byte, reencKey string) ([]byte, error) {
+	if config.Config.Reencrypt.Host == "" {
+		return nil, fmt.Errorf("host of the reencrypt service not set")
+	} else if config.Config.Reencrypt.Port == 0 {
+		return nil, fmt.Errorf("port of the reencrypt service not set")
+	}
+	address := fmt.Sprintf("%s:%d", config.Config.Reencrypt.Host, config.Config.Reencrypt.Port)
+	log.Debugf("Address of the reencrypt service: %s", address)
+
+	opts := []grpc.DialOption{
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	}
```

Comment:
This needs to be expanded so it can take credentials, user-pass and/or client certificates